### PR TITLE
Moved cohort module assignments to separate store. Updated module/ endpoints to support cohort and participant usage

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2334** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2334?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3  dependent PRs ([#2335](https://github.com/terraware/terraware-server/pull/2335) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2335?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2337](https://github.com/terraware/terraware-server/pull/2337) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2337?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2338](https://github.com/terraware/terraware-server/pull/2338) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2338?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#2332** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2332?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2333](https://github.com/terraware/terraware-server/pull/2333) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2333?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2331** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2330** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2330?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/db/CohortModuleStore.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  }
+
+  private fun fetch(
+      condition: Condition? = null,
+      joinForVisibility: (SelectOnConditionStep<Record>) -> SelectOnConditionStep<Record>
+  ): List<ModuleModel> {
+    val deliverablesField = deliverablesMultiset()
+
+    return with(MODULES) {
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/db/CohortModuleStore.kt]: Doesn't seem like the `with` is being used here.

**Review by sgrimm — APPROVED** [review]
This all makes sense. Nice tests!

**Comment by tommylau523** [comment]
### Merge activity

* **Aug 30, 2:06 PM EDT**: @tommylau523 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2334).
* **Aug 30, 2:06 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2334).

**Pull request merged**